google/auth/compute_engine/credentials.py

```diff
@@ -162,39 +162,52 @@ def __init__(
                 ignored.
             use_metadata_identity_endpoint (bool): Whether to use GCE metadata
                 identity endpoint. For backward compatibility the default value
-                is False. If set to True, ``request`` argument and all other
-                keyword arguments (``token_uri``, ``additional_claims``,
-                ``service_account_email``, ``signer``) will be ignored.
+                is False. If set to True, ``token_uri``, ``additional_claims``,
+                ``service_account_email``, ``signer`` argument should not be set;
+                otherwise ValueError will be raised.
+
+        Raises:
+            ValueError:
+                If ``use_metadata_identity_endpoint`` is set to True, and one of
+                ``token_uri``, ``additional_claims``, ``service_account_email``,
+                 ``signer`` arguments is set.
         """
         super(IDTokenCredentials, self).__init__()
 
-        if use_metadata_identity_endpoint or service_account_email is None:
-            sa_info = _metadata.get_service_account_info(request)
-            service_account_email = sa_info["email"]
-        self._service_account_email = service_account_email
-
-        self._target_audience = target_audience
-        self._additional_claims = {}
         self._use_metadata_identity_endpoint = use_metadata_identity_endpoint
+        self._target_audience = target_audience
 
         if use_metadata_identity_endpoint:
-            # If metadata identity endpoint is used, ignore all other keyword
-            # arguments
+            if token_uri or additional_claims or service_account_email or signer:
+                raise ValueError(
+                    "If use_metadata_identity_endpoint is set, token_uri, "
+                    "additional_claims, service_account_email, signer arguments"
+                    " must not be set"
+                )
             self._token_uri = None
+            self._additional_claims = None
             self._signer = None
+
+        if service_account_email is None:
+            sa_info = _metadata.get_service_account_info(request)
+            self._service_account_email = sa_info["email"]
         else:
+            self._service_account_email = service_account_email
+
+        if not use_metadata_identity_endpoint:
             if signer is None:
                 signer = iam.Signer(
                     request=request,
                     credentials=Credentials(),
-                    service_account_email=service_account_email,
+                    service_account_email=self._service_account_email,
                 )
             self._signer = signer
-
-            self._token_uri = token_uri
+            self._token_uri = _DEFAULT_TOKEN_URI
```

Comment:
Doesn't this change the previous behavior? It should be:

```py
self._token_uri = token_uri or _DEFAULT_TOKEN_URI
```

Reply:
Oops, thank you for the good catch! I fixed this and added unit test for token_uri.